format-xls/src/main/java/io/cdap/plugin/format/xls/input/XlsInputFormatUtils.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.format.xls.input;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Utilities around XLS input format.
+ */
+public class XlsInputFormatUtils {
+  private static final Pattern NOT_VALID_PATTERN = Pattern.compile("[^A-Za-z0-9_]+");
+
+  /**
+   * Cleans a list of column names to make sure they comply with avro field naming standard.
+   * It also makes sure each name is unique in the list.
+   * Field names can start with [A-Za-z_] and subsequently contain only [A-Za-z0-9_].
+   * <p>
+   * Steps:
+   * 1) Trim surrounding spaces
+   * 2) If its empty replace it with BLANK
+   * 3) If it starts with a number, prepend "col_"
+   * 4) Replace invalid characters with "_" (multiple invalid characters gets replaced with one symbol)
+   * 5) Check if the name has been found before (without considering case)
+   * if so add _# where # is the number of times seen before + 1
+   */
+  public static List<String> getSafeColumnNames(List<String> columnNames) {
```

Comment:
it doesn't look like the other formats do this column name manipulation. XLS should be consistent and avoid this logic. Leave it to Wrangler to cleanse column names.

Reply:
This idea was taken from csv format, this is required to get unique column names from xls when doing schema detection.

Reply:
I don't see where the csv format is doing this, can you point it out? 

If existing formats are doing this, then they should all share the same code instead of it being duplicated.

Reply:
You can find it under `package io.cdap.plugin.format.delimited.common`
class `DataTypeDetectorUtils` 
Method `setColumnNames` / `cleanSchemaFieldNames`

Reply:
was cleanSchemaColumnNames copied directly from there except modified to use Lists instead of arrays?

If the logic is the same, can you move that class to format-common to re-use most of the logic?